src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -148,20 +154,45 @@ protected ExtensionsRunner(Extension extension) throws IOException {
             .put(TransportSettings.PORT.getKey(), extensionSettings.getHostPort())
             .build();
         this.threadPool = new ThreadPool(settings);
+        this.taskManager = new TaskManager(settings, threadPool, Collections.emptySet());
 
-        Guice.createInjector(b -> {
+        // TODO: Move this map instantiation inside TransportActions and provide register API
+        // to move logic from the module stream
+        // https://github.com/opensearch-project/opensearch-sdk-java/issues/467
+        Map<String, Class<? extends TransportAction<? extends ActionRequest, ? extends ActionResponse>>> transportActionsMap =
+            new HashMap<>();
+
+        List<com.google.inject.Module> modules = new ArrayList<>();
+        // Base bindings
+        modules.add(b -> {
             b.bind(ExtensionsRunner.class).toInstance(this);
             b.bind(Extension.class).toInstance(extension);
 
+            // FIXME: Change this to a provider interface
+            // https://github.com/opensearch-project/opensearch-sdk-java/issues/447
             b.bind(NamedXContentRegistry.class).toInstance(getNamedXContentRegistry().getRegistry());
             b.bind(ThreadPool.class).toInstance(getThreadPool());
+            b.bind(TaskManager.class).toInstance(taskManager);
 
-            b.bind(SDKClient.class).toInstance(new SDKClient());
+            b.bind(SDKClient.class);
             b.bind(SDKClusterService.class).toInstance(new SDKClusterService(this));
-
-            // create components
-            injectComponents(b);
         });
+        // Bind the return values from create components
+        modules.add(this::injectComponents);
+        // Bind actions from getActions
+        if (extension instanceof ActionExtension) {
+            SDKActionModule sdkActionModule = new SDKActionModule((ActionExtension) extension);
+            modules.add(sdkActionModule);
+            // save custom transport actions
+            sdkActionModule.getActions()
+                .entrySet()
+                .stream()
+                .forEach(h -> { transportActionsMap.put(h.getKey(), h.getValue().getTransportAction()); });
+        }
```

Comment:
No because it's part of the `TransportActions` class which isn't available there.

But this is just a hack for now, better idea is in #467, see the TODO for that `transportActionsMap` definition above!